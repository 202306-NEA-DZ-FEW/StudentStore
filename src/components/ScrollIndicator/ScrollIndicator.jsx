import React, { useEffect, useState } from "react";

const ScrollIndicator = () => {
    const [scrollTop, setScrollTop] = useState(0);

    const onScroll = () => {
        const winScroll =
            document.body.scrollTop || document.documentElement.scrollTop;
        const height =
            document.documentElement.scrollHeight -
            document.documentElement.clientHeight;

        const scrolled = (winScroll / height) * 100;

        setScrollTop(scrolled);
    };

    useEffect(() => {
        window.addEventListener("scroll", onScroll);
        return () => {
            window.removeEventListener("scroll", onScroll);
        };
    }, []);

    return (
        <div className='scroll-indicator'>
            <div className='progressMainWrapper bg-indigo- h-[6px] mt-0 ml-0 z-1 w-full'>
                <div
                    className='progressMainStyle bg-orange-500 h-[6px] mt-0 ml-0 z-1 w-15'
                    style={{ width: `${scrollTop}%` }}
                ></div>
            </div>
        </div>
    );
};

export default ScrollIndicator;
