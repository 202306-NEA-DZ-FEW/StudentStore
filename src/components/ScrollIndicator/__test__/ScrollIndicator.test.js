import renderer from "react-test-renderer";

import ScrollIndicator from "../ScrollIndicator";

it("renders correctly", () => {
    const tree = renderer.create(<ScrollIndicator />).toJSON();
    expect(tree).toMatchSnapshot();
});
