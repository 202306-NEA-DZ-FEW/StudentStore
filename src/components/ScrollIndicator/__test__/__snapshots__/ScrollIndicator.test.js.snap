// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="scroll-indicator"
>
  <div
    className="progressMainWrapper bg-indigo- h-[6px] mt-0 ml-0 z-1 w-full"
  >
    <div
      className="progressMainStyle bg-orange-500 h-[6px] mt-0 ml-0 z-1 w-15"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
</div>
`;
